### PR TITLE
Class22 127 remove favorites confirmation via modal

### PR DESCRIPTION
**Pull request by patel-prakashkumar**
# Description:

# Jira:

### CODE REVIEW

File: packages/client/src/components/ProductCard/ProductCard.component.js [hunk 1]
```diff
@@ -7,21 +7,33 @@ import StarRateIcon from '../../assets/star-rate-icon.svg';
...
+                ? setOpenModal({ modalStatus: true, favoriteId: id })
+                : toggleFavorite(id, title, price, event);
+            }}
             type="button"
             className="favorite-button"
           >
             <img
               src={isFavorite ? FullFavoriteIcon : EmptyFavoriteIcon}
               alt="Favorite Icon"
+              role="presentation"
```
Comment (miltonhultgren) [inline packages/client/src/components/ProductCard/ProductCard.component.js]: Not sure why you're adding this? I would avoid changing ARIA roles unless you really know what you are doing.

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js [hunk 2]
```diff
@@ -3,16 +3,26 @@ import SearchInput from '../../components/SearchInput/SearchInput.component';
 import { getLocalStorage, setLocalStorage } from '../../utils/storageHelpers';
 import ProductCard from '../../components/ProductCard/ProductCard.component';
 import './FavoritesPage.style.css';
+import Modal from '../../components/Modal/Modal.component';
+import { ViewPageButton } from '../../components/ViewPageButton/ViewPageButton.component';
 
 export const FavoritesPage = () => {
+  const [openModal, setOpenModal] = useState({
```
Comment (miltonhultgren) [inline packages/client/src/containers/FavoritesPage/FavoritesPage.container.js]: `modalState` might be a better name here.

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js [hunk 3]
```diff
@@ -3,16 +3,26 @@ import SearchInput from '../../components/SearchInput/SearchInput.component';
...
+    modalStatus: false,
+    favoriteId: -1,
+  });
+  const toggleModal = () => {
+    setOpenModal({ modalStatus: false, favoriteId: -1 });
+    document.body.style.overflow = 'visible';
```
Comment (miltonhultgren) [inline packages/client/src/containers/FavoritesPage/FavoritesPage.container.js]: In general you should never modify the CSS of the page directly, instead you should alter this with a CSS class that is added within the React markup instead.

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js [hunk 4]
```diff
@@ -3,16 +3,26 @@ import SearchInput from '../../components/SearchInput/SearchInput.component';
...
 import './FavoritesPage.style.css';
+import Modal from '../../components/Modal/Modal.component';
+import { ViewPageButton } from '../../components/ViewPageButton/ViewPageButton.component';
 
 export const FavoritesPage = () => {
+  const [openModal, setOpenModal] = useState({
+    modalStatus: false,
+    favoriteId: -1,
+  });
+  const toggleModal = () => {
```
Comment (miltonhultgren) [inline packages/client/src/containers/FavoritesPage/FavoritesPage.container.js]: closeModal?

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js [hunk 5]
```diff
@@ -34,16 +43,42 @@ export const FavoritesPage = () => {
...
           id={product.id}
+          setOpenModal={setOpenModal}
           toggleFavorite={toggleFavorite}
           isFavorite={favoriteProducts.some((x) => x.id === product.id)}
         />
       </div>
     );
   });
 
+  const handleModal = (favoriteId) => {
```
Comment (miltonhultgren) [inline packages/client/src/containers/FavoritesPage/FavoritesPage.container.js:53]: removeFavorite?

**Review by miltonhultgren — CHANGES_REQUESTED** [review]

File: packages/client/src/components/ProductCard/ProductCard.component.js (hunk 1 above)
Comment (27ManishaPatel) [reply packages/client/src/components/ProductCard/ProductCard.component.js]: That is the element for screen readers to understand but it should not be here so I will remove it, Thank you 😊

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js (hunk 3 above)
Comment (27ManishaPatel) [reply packages/client/src/containers/FavoritesPage/FavoritesPage.container.js]: True I can remove it

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js (hunk 4 above)
Comment (27ManishaPatel) [reply packages/client/src/containers/FavoritesPage/FavoritesPage.container.js]: Yes it is closeModel

File: packages/client/src/containers/FavoritesPage/FavoritesPage.container.js (hunk 5 above)
Comment (27ManishaPatel) [reply packages/client/src/containers/FavoritesPage/FavoritesPage.container.js:53]: Yes handleModal will remove it

**Pull request merged**